src/main/java/ch/njol/skript/expressions/ExprBossBar.java

```diff
@@ -0,0 +1,122 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Color;
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.DyeColor;
+import org.bukkit.boss.BarColor;
+import org.bukkit.boss.BarStyle;
+import org.bukkit.boss.BossBar;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Boss Bar")
+@Description("""
+	A new boss bar.
+	Boss bars have several properties (a title/name, a color, a style, visibility and flags).
+	For players to see the boss bar they must be added to it. Players may be added to multiple boss bars at a time.""")
+@Examples({
+	"set {bar} to a new boss bar",
+	"set the name of {bar} to \"hello\"",
+	"set the color of {bar} to red",
+	"add player to {bar}",
+	"set {bar} to a new pink boss bar named \"hello\""
+})
+@Since("INSERT VERSION")
+public class ExprBossBar extends SimpleExpression<BossBar> {
+
+	static {
+		Skript.registerExpression(ExprBossBar.class, BossBar.class, ExpressionType.SIMPLE,
```

Comment:
```suggestion
		Skript.registerExpression(ExprBossBar.class, BossBar.class, ExpressionType.COMBINED,
```
Generally fits better with COMBINED